kubeadm: rename flag to --ignore-preflight-errors

kad opened the pull request:
**What this PR does / why we need it**:
Addresses comments in #56072
Improves user experience by using name that is more
descriptive.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#480

**Special notes for your reviewer**:
/area kubeadm
/sig cluster-lifecycle
/assign @timothysc

**Release note**:
```release-note
NONE
```

kad committed:
kubeadm: rename flag to --ignore-preflight-errors

Improves user experience by using name that is more
descriptive.

luxas reviewed (commented):
As said in the referenced PR, please update variable names as well for consistency

kad commented:
@luxas updated with variable and function names changed.

xiangpengzhao commented:
Just a gofmt verification error now :)
<details>

```
I1122 15:13:28.316] Verifying hack/make-rules/../../hack/verify-gofmt.sh
I1122 15:13:42.480] diff -u ./cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go.orig ./cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go
I1122 15:13:42.480] --- ./cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go.orig	2017-11-22 15:13:34.992778641 +0000
I1122 15:13:42.480] +++ ./cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go	2017-11-22 15:13:34.992778641 +0000
I1122 15:13:42.480] @@ -461,10 +461,10 @@
I1122 15:13:42.481]  
I1122 15:13:42.481]  func TestValidateIgnorePreflightErrors(t *testing.T) {
I1122 15:13:42.481]  	var tests = []struct {
I1122 15:13:42.481] -		ignorePreflightErrors  []string
I1122 15:13:42.481] -		skipPreflightChecks bool
I1122 15:13:42.481] -		expectedLen         int
I1122 15:13:42.482] -		expectedError       bool
I1122 15:13:42.482] +		ignorePreflightErrors []string
I1122 15:13:42.482] +		skipPreflightChecks   bool
I1122 15:13:42.482] +		expectedLen           int
I1122 15:13:42.482] +		expectedError         bool
I1122 15:13:42.482]  	}{
I1122 15:13:42.482]  		{[]string{}, false, 0, false},                             // empty list, no old skip-preflight-checks
I1122 15:13:42.482]  		{[]string{}, true, 1, false},                              // empty list, old skip-preflight-checks
I1122 15:13:42.483] diff -u ./cmd/kubeadm/app/cmd/upgrade/upgrade.go.orig ./cmd/kubeadm/app/cmd/upgrade/upgrade.go
I1122 15:13:42.483] --- ./cmd/kubeadm/app/cmd/upgrade/upgrade.go.orig	2017-11-22 15:13:35.168778732 +0000
I1122 15:13:42.483] +++ ./cmd/kubeadm/app/cmd/upgrade/upgrade.go	2017-11-22 15:13:35.168778732 +0000
I1122 15:13:42.483] @@ -32,8 +32,8 @@
I1122 15:13:42.483]  	allowRCUpgrades           bool
I1122 15:13:42.483]  	printConfig               bool
I1122 15:13:42.483]  	skipPreFlight             bool
I1122 15:13:42.483] -	ignorePreflightErrors        []string
I1122 15:13:42.484] -	ignorePreflightErrorsSet     sets.String
I1122 15:13:42.484] +	ignorePreflightErrors     []string
I1122 15:13:42.484] +	ignorePreflightErrorsSet  sets.String
I1122 15:13:42.484]  }
I1122 15:13:42.484]  
I1122 15:13:42.484]  // NewCmdUpgrade returns the cobra command for `kubeadm upgrade`
I1122 15:13:42.484] @@ -45,7 +45,7 @@
I1122 15:13:42.484]  		allowRCUpgrades:           false,
I1122 15:13:42.485]  		printConfig:               false,
I1122 15:13:42.485]  		skipPreFlight:             false,
I1122 15:13:42.485] -		ignorePreflightErrorsSet:     sets.NewString(),
I1122 15:13:42.485] +		ignorePreflightErrorsSet:  sets.NewString(),
I1122 15:13:42.485]  	}
I1122 15:13:42.485]  
I1122 15:13:42.485]  	cmd := &cobra.Command{
I1122 15:13:42.485] FAILED   hack/make-rules/../../hack/verify-gofmt.sh	14s
```

</details>

kad commented:
Yep, edited in simple vi instead of vim with plugins :(

luxas commented on cmd/kubeadm/app/apis/kubeadm/validation/validation.go:
```diff
@@ -331,25 +331,25 @@ func ValidateAPIEndpoint(c *kubeadm.MasterConfiguration, fldPath *field.Path) fi
 	if skipPreflightChecks {
 		if ignoreErrors.Has("all") {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-checks-errors"), strings.Join(ignoreErrors.List(), ","), "'all' is used together with deprecated flag --skip-preflight-checks. Remove deprecated flag"))
+			allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-preflight-errors"), strings.Join(ignoreErrors.List(), ","), "'all' is used together with deprecated flag --skip-preflight-checks. Remove deprecated flag"))
```
ref: https://github.com/kubernetes/kubernetes/pull/56072#pullrequestreview-78561104

luxas reviewed (approved):
/lgtm
/approve no-issue

jberkus commented:
/priority critical-urgent

/remove-priority important-soon

adjusting priorities for code freeze

kad commented:
Rebased together with addressing few more comments from @luxas:
- removed error scenario if old and new flags are used at the same time. Deprecation warning is still produced for old flag.
- added unit test for `setHasItemOrAll`

xiangpengzhao commented:
/lgtm

/test pull-kubernetes-e2e-gce-device-plugin-gpu

kad commented:
/test pull-kubernetes-unit

xiangpengzhao commented:
/test pull-kubernetes-e2e-gce
